### PR TITLE
feat(core): remove entryComponents

### PR DESCRIPTION
**Pull request by crisbeto**
`entryComponents` have been deprecated since version 9, because with Ivy they weren't necessary. These changes remove any remaining references.

BREAKING CHANGE:
* `entryComponents` has been deleted from the `@NgModule` and `@Component` public APIs. Any usages can be removed since they weren't doing anyting.
* `ANALYZE_FOR_ENTRY_COMPONENTS` injection token has been deleted. Any references can be removed.

### CODE REVIEW

**Review by alxhub — APPROVED** [review]

File: aio/content/guide/deprecations.md
Comment (JeanMeche) [inline aio/content/guide/deprecations.md:1]: Is it conscious choice to have kept the entry and the section about the `entryComponents` & `ANALYZE_FOR_ENTRY_COMPONENTS` ?

Comment (crisbeto) [reply aio/content/guide/deprecations.md:1]: Yes, I wanted to keep the reference in case folks see an error don't know where it's coming from.

**Review by mgechev — APPROVED** [review]

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: public-api

**Comment by crisbeto** [comment]
[Passing TGP](http://test/OCL:517940697:BASE:518206399:1679389100285:68d410b3)

**Comment by atscott** [comment]
This PR was merged into the repository by commit 585e34bf6c86f7b056b0aafaaca056baedaedae3.

**Pull request closed without merging**